src/v/datalake/coordinator/BUILD

```diff
@@ -0,0 +1,42 @@
+load("//bazel:build.bzl", "redpanda_cc_library")
+load("//src/v/rpc:compiler.bzl", "redpanda_cc_rpc_library")
+
+redpanda_cc_rpc_library(
+    name = "generated_datalake_coordinator_rpc",
+    src = "rpc.json",
+    include_prefix = "datalake/coordinator",
+)
+
+redpanda_cc_library(
+    name = "coordinator",
+    srcs = [
+        "frontend.cc",
+        "service.cc",
+        "state_machine.cc",
+    ],
+    hdrs = [
+        "frontend.h",
+        "service.h",
+        "state_machine.h",
+        "types.h",
+    ],
```

Comment:
Not suggesting to change it, but any reason these are bundled together?

Reply:
I've split it further.